### ISSUE TITLE
withCount() aggregate accessors not resolved as magic properties

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#503]
## Description

When an Eloquent model uses `withCount()` (e.g., `$query->withCount('contacts')`), Laravel adds a magic `{relation}_count` attribute (e.g., `$model->contacts_count`) as an integer. The plugin does not recognize these aggregate accessors, causing `UndefinedMagicPropertyFetch` at Psalm errorLevel 1.

## Impact

**10 false positives** across [monicahq/monica](https://github.com/monicahq/monica) (262 project files).

Affected patterns:
- `App\Models\Label::$contacts_count` (3 occurrences)
- `App\Models\Address::$contacts_count` (2)
- `App\Models\Vault::$contacts_count` (1)
- `App\Models\Tag::$posts_count` (1)
- and others

## Reproduction

```php
// Query
$labels = Label::withCount('contacts')->get();

// Usage — triggers UndefinedMagicPropertyFetch
foreach ($labels as $label) {
    echo $label->contacts_count; // int
}
```

## Expected Behavior

The plugin should recognize `{relation}_{function}` properties as valid magic attributes when the model has a corresponding relationship method.

## Full list of aggregate methods

All these methods (from [Laravel docs](https://laravel.com/docs/12.x/eloquent-relationships#other-aggregate-functions)) add magic attributes following the `{relation}_{function}` naming convention. All delegate to `withAggregate()` / `loadAggregate()` internally.

### Query builder methods (`QueriesRelationships` trait)

| Method | Signature | Attribute suffix | Attribute type |
|--------|-----------|-----------------|----------------|
| `withCount` | `withCount($relations)` | `_count` | `int` |
| `withMin` | `withMin($relation, $column)` | `_min` | `string\|null` |
| `withMax` | `withMax($relation, $column)` | `_max` | `string\|null` |
| `withSum` | `withSum($relation, $column)` | `_sum` | `numeric-string\|null` |
| `withAvg` | `withAvg($relation, $column)` | `_avg` | `numeric-string\|null` |
| `withExists` | `withExists($relation)` | `_exists` | `bool` |
| `withAggregate` | `withAggregate($relations, $column, $function = null)` | `_{function}` | varies |

### Model instance methods (`Model` class)

| Method | Signature | Attribute suffix | Attribute type |
|--------|-----------|-----------------|----------------|
| `loadCount` | `loadCount($relations)` | `_count` | `int` |
| `loadMin` | `loadMin($relations, $column)` | `_min` | `string\|null` |
| `loadMax` | `loadMax($relations, $column)` | `_max` | `string\|null` |
| `loadSum` | `loadSum($relations, $column)` | `_sum` | `numeric-string\|null` |
| `loadAvg` | `loadAvg($relations, $column)` | `_avg` | `numeric-string\|null` |
| `loadExists` | `loadExists($relations)` | `_exists` | `bool` |
| `loadAggregate` | `loadAggregate($relations, $column, $function = null)` | `_{function}` | varies |

### Notes on types

- `withCount`/`loadCount` returns `int` (SQL `COUNT(*)` cast to integer by Laravel)
- `withExists`/`loadExists` returns `bool` (SQL `EXISTS` cast to boolean by Laravel)
- `withMin`/`withMax`/`loadMin`/`loadMax` return `string|null` (raw database value, not cast)
- `withSum`/`withAvg`/`loadSum`/`loadAvg` return `numeric-string|null` (raw database value, not cast)
- Custom aliases via `withAggregate` follow the same `{relation}_{function}` pattern

## Proposed approach

The simplest heuristic: if a model has a relationship method `contacts()`, then `contacts_count`, `contacts_sum`, `contacts_min`, `contacts_max`, `contacts_avg`, and `contacts_exists` are all valid magic properties. The `ModelPropertyHandler` or `ModelRelationshipPropertyHandler` could recognize the `_{suffix}` pattern and resolve the type based on the suffix.

## Context

Found while auditing [monicahq/monica](https://github.com/monicahq/monica). After #502 (relationship accessor resolution), these are the largest remaining source of Eloquent-related `UndefinedMagicPropertyFetch`.
